common/protob/messages-cardano.proto

```diff
@@ -500,3 +500,49 @@ message CardanoTxBodyHash {
  */
 message CardanoSignTxFinished {
 }
+
+/**
+ * Request: Ask device to sign a message containing arbitrary data
+ * @start
+ * @next CardanoMessageItemAck
+ */
+message CardanoSignMessageInit {
+    required uint32 protocol_magic = 1;                           // network's protocol magic
+    required uint32 network_id = 2;                               // network id - mainnet or testnet
+    repeated uint32 signing_path = 3;                             // BIP-32-style path to derive the signing key from master node
+    required uint32 payload_size = 4;                             // size of the payload to be signed
+    required bool hash_payload = 5;                               // whether to hash the payload before signing
+    repeated uint32 key_path = 6;                                 // mutually exclusive with address_parameters
```

Comment:
This field seems not really useful --- if we use key hash for the address field, it will be the key used to sign the msg, so it seems we can just use `signing_path`?

Reply:
I assumed it was implied by this part of the doc that we want to separate the two paths.
> Since the address field might be unrelated to the key used for signing and it seems impossible to completely determine what the existing applications use, we don’t want to impose restrictions on the derivation paths used.

Did I understand it wrong?

Reply:
The sentence you quote is somewhat ambiguous. What I meant was that in general according to CIP-8 it might be unrelated and some historical use could be there _in case of address_. The usage of just key hash in the address field is new (AFAIK --- discussed with Ryan, gitmatchtl etc.), so we do not need to support arbitrary keys different from signing keys there. I'd be OK if you kept it implemented like this (if it works now). What makes sense to me is not to display the key path twice if it is the same --- i.e. if there is just a key hash in the address field and the key is used for signing, only show that once (I'm not sure at which place, because I don't know how people use the arbitrary msg signing feature, but I'd perhaps put more emphasis on the key that is being used to sign and always explicitly say what that key path is, and hide info about the address field from the user if it does not say anything new.)

Reply:
I guess I won't add "arbitrary key in addr field (different from signing key)" to Ledger because it would cut into APDU space. I'll see today/tomorrow.